dotnet/Capstone/Controllers/BirdListController.cs

```diff
@@ -0,0 +1,141 @@
+﻿using Capstone.DAO;
+using Capstone.Exceptions;
+using Capstone.Models;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Security.Principal;
+
+namespace Capstone.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class BirdListController: ControllerBase
+    {
+        private readonly IBirdListDao birdListDao;
+
+        public BirdListController(IBirdListDao birdListDao)
+        {
+            this.birdListDao = birdListDao;
+        }
+
+
+
+        [HttpGet("/lists/{id}")]
+        public BirdList getList(int id)
+        {
+            BirdList birdList = birdListDao.getList(id);
+
+            if (birdList == null)
+            {
+                Console.WriteLine("No bird matches this id");
+            }
+
+            return birdList;
+
+        }
+
+
+        [HttpGet("/lists")]
+        public List<BirdList> getAllLists()
+        {
+                
+            List<BirdList> birdList = birdListDao.getAllLists();
+
+            if (birdList == null)
+            {
+                Console.WriteLine("No lists");
+            }
+
+            return birdList;
+        }
+
+
+        [HttpPut("/editList/{id}")]
+        public IActionResult editList(BirdList list, int id)
+        {
+            const string errorMessage = "An error occurred and list could not be modified.";
+            
+            IActionResult result;
+
+            try
+            {
+                birdListDao.editList(list, id);
+                result = Ok();
+            }
+            catch(ArgumentException e)
+            {
+                result = BadRequest(new {message = e.Message});
+            }
+            catch
+            {
+                result = BadRequest(new { message = errorMessage });
+            }
+            return result;
+
+        }
+
+
+        [HttpDelete("/deleteList/{id}")]
```

Comment:
This should either send an error message if the list isn't empty or delete all birds in the list as well. 

Also, I haven't tested it, but it looks like there's no reason a user couldn't delete another user's list.

Reply:
Done. 
A user cannot delete  another user's list because those list are not available to them anymore.